bundles/org.openhab.binding.onecta/src/main/java/org/openhab/binding/onecta/internal/handler/OnectaBridgeHandler.java

```diff
@@ -0,0 +1,167 @@
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initialize.");
+        config = getConfigAs(OnectaConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            onectaConnectionClient.startConnecton(thing.getConfiguration().get(CONFIG_PAR_USERID).toString(),
+                    thing.getConfiguration().get(CONFIG_PAR_PASSWORD).toString());
+
+            if (onectaConnectionClient.isOnline()) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } catch (DaikinCommunicationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollDevices, 10,
+                Integer.parseInt(thing.getConfiguration().get(CONFIG_PAR_REFRESHINTERVAL).toString()),
+                TimeUnit.SECONDS);
+
+        // Trigger discovery of Devices
+        scheduler.submit(runnable);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+    }
+
+    private void pollDevices() {
+        logger.debug("pollDevices.");
+
+        if (getThing().getStatus().equals(ThingStatus.OFFLINE)) {
+            try {
+                logger.debug("Try to restore connection ");
+                onectaConnectionClient.restoreConnecton();
+
+                if (onectaConnectionClient.isOnline()) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (DaikinCommunicationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Try to restore connection. See log for more information. ");
+            }
+        }
+
+        if (getThing().getStatus().equals(ThingStatus.ONLINE)) {
+
+            try {
+                onectaConnectionClient.refreshUnitsData();
+            } catch (DaikinCommunicationException e) {
+                logger.debug("DaikinCommunicationException: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+
+            List<Thing> things = getThing().getThings();
+            for (Thing t : things) {
+                // BaseThingHandler handler;
+                if (t.getStatus().equals(ThingStatus.ONLINE)) {
+
+                    if (t.getThingTypeUID().equals(THING_TYPE_CLIMATECONTROL)) {
+                        OnectaDeviceHandler onectaDeviceHandler = (OnectaDeviceHandler) t.getHandler();
```

Comment:
If all these handlers have an shared interface, this code could be simplified by casting the t.getHandler() to that interface and call refreshdevice. as this is the same for all channels.